util/wait_group.go

```diff
@@ -0,0 +1,34 @@
+// Copyright 2021 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package util
+
+import (
+	"sync"
+	"time"
+)
+
+func WaitTimeout(wg *sync.WaitGroup, timeout time.Duration) bool {
```

Comment:
When someone want the "wait with timeout feature", why don't they use channel directly?
That will not hide the cost of new goroutine and the allocation of a new channel etc.

Reply:
How to use channel directly?

Reply:
I think I got it, let me refactor.

Reply:
Rafactored and removed.